controllers/datadogagent/feature/liveprocess/feature_test.go

```diff
@@ -0,0 +1,141 @@
+	ddav1LiveProcessDisabled := v1alpha1.DatadogAgent{
+		Spec: v1alpha1.DatadogAgentSpec{
+			Agent: v1alpha1.DatadogAgentSpecAgentSpec{
+				Process: &v1alpha1.ProcessSpec{
+					ProcessCollectionEnabled: apiutils.NewBoolPointer(false),
+				},
+			},
+		},
+	}
+
+	ddav1LiveProcessEnabled := ddav1LiveProcessDisabled.DeepCopy()
+	{
+		ddav1LiveProcessEnabled.Spec.Agent.Process.ProcessCollectionEnabled = apiutils.NewBoolPointer(true)
+	}
+
+	ddav2LiveProcessDisabled := v2alpha1.DatadogAgent{
+		Spec: v2alpha1.DatadogAgentSpec{
+			Features: &v2alpha1.DatadogFeatures{
+				LiveProcessCollection: &v2alpha1.LiveProcessCollectionFeatureConfig{
+					Enabled: apiutils.NewBoolPointer(false),
+				},
+			},
+		},
+	}
+	ddav2LiveProcessEnabled := ddav2LiveProcessDisabled.DeepCopy()
+	{
+		ddav2LiveProcessEnabled.Spec.Features.LiveProcessCollection.Enabled = apiutils.NewBoolPointer(true)
+	}
+
+	liveProcessAgentNodeWantFunc := func(t testing.TB, mgrInterface feature.PodTemplateManagers) {
+		mgr := mgrInterface.(*fake.PodTemplateManagers)
+
+		// check volume mounts
+		wantVolumeMounts := []corev1.VolumeMount{
+			{
+				Name:      apicommon.PasswdVolumeName,
+				MountPath: apicommon.PasswdMountPath,
+				ReadOnly:  true,
+			},
+		}
+
+		processAgentMounts := mgr.VolumeMountMgr.VolumeMountsByC[apicommonv1.ProcessAgentContainerName]
+		assert.True(t, apiutils.IsEqualStruct(processAgentMounts, wantVolumeMounts), "Process Agent volume mounts \ndiff = %s", cmp.Diff(processAgentMounts, wantVolumeMounts))
+
+		// check volumes
+		wantVolumes := []corev1.Volume{
+			{
+				Name: apicommon.PasswdVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: apicommon.PasswdHostPath,
+					},
+				},
+			},
+		}
+
+		volumes := mgr.VolumeMgr.Volumes
+		assert.True(t, apiutils.IsEqualStruct(volumes, wantVolumes), "Volumes \ndiff = %s", cmp.Diff(volumes, wantVolumes))
+
+		// check env vars
+		wantEnvVars := []*corev1.EnvVar{
+			{
+				Name:  apicommon.DDProcessAgentEnabled,
+				Value: "true",
+			},
+		}
+		agentEnvVars := mgr.EnvVarMgr.EnvVarsByC[apicommonv1.CoreAgentContainerName]
+		assert.True(t, apiutils.IsEqualStruct(agentEnvVars, wantEnvVars), "Agent envvars \ndiff = %s", cmp.Diff(agentEnvVars, wantEnvVars))
+
+		processAgentEnvVars := mgr.EnvVarMgr.EnvVarsByC[apicommonv1.ProcessAgentContainerName]
+		assert.True(t, apiutils.IsEqualStruct(processAgentEnvVars, wantEnvVars), "Process Agent envvars \ndiff = %s", cmp.Diff(processAgentEnvVars, wantEnvVars))
+	}
+
+	tests := test.FeatureTestSuite{
+		///////////////////////////
+		// v1alpha1.DatadogAgent //
+		///////////////////////////
+		{
+			Name:          "v1alpha1 live process collection not enabled",
+			DDAv1:         ddav1LiveProcessDisabled.DeepCopy(),
+			WantConfigure: false,
+		},
+		{
+			Name:          "v1alpha1 live process collection enabled",
+			DDAv1:         ddav1LiveProcessEnabled,
+			WantConfigure: true,
+			Agent: &test.ComponentTest{
```

Comment:
Nit: you can use the new helpers that we merged some days ago to avoid duplicating some code:

https://github.com/DataDog/datadog-operator/blob/377f96fe19a6a1fd4a9d54590e814fd1065a7944/controllers/datadogagent/feature/test/testsuite.go#L66

With that, you can remove the createEmptyFakeManager function defined at the top of this file, and here use: `Agent: test.NewDefaultComponentTest().WithWantFunc(liveProcessAgentNodeWantFunc)`. Same for the test below.